tests/features/test_cuda_interface.py

```diff
@@ -0,0 +1,64 @@
+import numpy as np
+import pytest
+import sys
+
+try:
+    import cupy as cp
```

Comment:
it would be good to:

1. Add cupy to the list of test dependencies at https://github.com/genn-team/genn/blob/master/setup.py#L316
2. Set the CI to install the dependency at https://github.com/genn-team/genn/blob/master/Jenkinsfile#L196